packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1472,20 +1549,29 @@ abstract class CustomPainter {
   bool hitTest(Point position) => null;
 }
 
-/// Delegates its painting
+/// Provides a canvas on which to draw during the paint phase.
 ///
-/// When asked to paint, custom paint first asks painter to paint with the
-/// current canvas and then paints its children. After painting its children,
-/// custom paint asks foregroundPainter to paint. The coodinate system of the
-/// canvas matches the coordinate system of the custom paint object. The
-/// painters are expected to paint within a rectangle starting at the origin
-/// and encompassing a region of the given size. If the painters paints outside
+/// When asked to paint, [CustomPaint] first ask its [painter] to paint on the
+/// current canvas, then it paints its child, and then, after painting its
+/// child, it ask its [foregroundPainter] to paint. The coodinate system of the
+/// canvas matches the coordinate system of the [CustomPaint] object. The
+/// painters are expected to paint within a rectangle starting at the origin and
+/// encompassing a region of the given size. (If the painters paints outside
 /// those bounds, there might be insufficient memory allocated to rasterize the
-/// painting commands and the resulting behavior is undefined.
+/// painting commands and the resulting behavior is undefined.)
+///
+/// Painters are implemented by subclassing [CustomPainter].
+///
+/// Because custom paint calls its painters during paint, you cannot mark the
+/// tree as needing a new layout during the callback (the layout for this frame
+/// has already happened).
 ///
-/// Because custom paint calls its painters during paint, you cannot dirty
-/// layout or paint information during the callback.
+/// See also:
+///
+///  * [CustomPainter],
```

Comment:
check what the usual punctuation is for see-also lists... i don't think we normally put a comma there